go/store/datas/puller.go

```diff
@@ -303,14 +302,23 @@ func limitToNewChunks(absent hash.HashSet, downloaded hash.HashSet) {
 	}
 }
 
+type nbsGetManyCompressedable interface {
+	GetManyCompressed(context.Context, hash.HashSet, chan<- nbs.CompressedChunk) error
+}
+
 func (p *Puller) getCmp(ctx context.Context, twDetails *TreeWalkEventDetails, leaves, batch hash.HashSet, completedTables chan FilledWriters) (hash.HashSet, hash.HashSet, error) {
-	found := make(chan chunks.Chunkable, 4096)
+	found := make(chan nbs.CompressedChunk, 4096)
 	processed := make(chan CmpChnkAndRefs, 4096)
 
 	ae := atomicerr.New()
 	go func() {
 		defer close(found)
-		err := p.srcDB.chunkStore().GetManyCompressed(ctx, batch, found)
+		st, ok := p.srcDB.chunkStore().(nbsGetManyCompressedable)
```

Comment:
maybe create a new interface that is both a ChunkStore and an nbsGetManyCompressedable:

    type CompressedableChunkStore interface {
        chunks.ChunkStore
        nbsGetManyCompressedable
    }

and then cache the chunkstores as members in the NewPuller call.  You can error there if the interface isn't implemented and then you don't have to wait til the code gets here to fail in this go routine.

Reply:
This was better. Thanks for the suggestion!